File: in_Java/0515 Find Largest Value in Each Tree Row.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Integer> largestValues(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        if(root == null) return res;
        //res.add(root.val);
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        int len;
        int curmax; 
        TreeNode cur;
        while(q.size() != 0){
            curmax = Integer.MIN_VALUE;
            len = q.size();
            while(len-- != 0){
                cur = q.poll();
                curmax = Math.max(curmax,cur.val);
                if(cur.left != null) q.add(cur.left);
                if(cur.right != null) q.add(cur.right);
            }
            res.add(curmax);
        }
        return res;
    }
}

File: in_Python/0515 Find Largest Value in Each Tree Row.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def largestValues(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        res = []
        if not root:
            return res
        q = [root]
        while q:
            n = len(q)
            curmax = -sys.maxint
            while n != 0:
                n -= 1
                h = q.pop(0)
                curmax = max(curmax, h.val)
                if h.left:
                    q.append(h.left)
                if h.right:
                    q.append(h.right)
            res.append(curmax)
        return res